task/07-yield-tasks.js

```diff
@@ -144,16 +211,28 @@ function* mergeSortedSequences(source1, source2) {
  *
  *   Most popular implementation of the logic in npm https://www.npmjs.com/package/co
  */
-function async(generator) {
-    throw new Error('Not implemented');
-}
 
+async function async(generator) {
+  let gen = generator(),
+    arg = null,
+    current;
+  while (true) {
```

Comment:
it's better to not use while(true) for finite loops and put exit condition in loop definition